server/src-lib/Hasura/Server/Init.hs

```diff
@@ -1,5 +1,6 @@
 -- | Types and functions related to the server initialisation
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE CPP    #-}
+{-# OPTIONS_GHC -O0 #-}
```

Comment:
I tried `-O1` too, seems to consume less memory but still ghc doesn't move past this module (gave it a good 5 mins).